modules/data.atmosphere/R/download.NOAA_GEFS_downscale.R

```diff
@@ -0,0 +1,341 @@
+##' NOAA.GEFS.willow creek.3.2018-06-08T06:00.2018-06-24T06:00.nc specifies the forecast, using ensemble nubmer 3 at willow creek on
+##' June 6th, 2018 at 6:00 a.m. to June 24th, 2018 at 6:00 a.m.
+##' 
+##' @return A list of data frames is returned containing information about the data file that can be used to locate it later.  Each
+##' data frame contains information about one file.
+##'
+##' @param outfolder Directory where results should be written
+##' @param start_date, end_date Range of dates/times to be downloaded (default assumed time of day is 0:00, midnight)
+##' @param lat site latitude in decimal degrees
+##' @param lon site longitude in decimal degrees
+##' @param sitename The unique ID given to each site. This is used as part of the file name.
+##' @param overwrite logical. Download a fresh version even if a local file with the same name already exists?
+##' @param verbose logical.  Print additional debug information.  Passed on to functions in the netcdf4 package to provide debugging info.
+##' @param ... Other arguments, currently ignored
+##' @export
+##' 
+##' @examples 
+##' \dontrun{
+##'  download.NOAA_GEFS(outfolder="~/Working/results", lat.in= 45.805925, lon.in = -90.07961, sitename="US-WCr")
+##' }
+##' 
+##' @author Katie Zarada - modified code from Luke Dramko and Laura Puckett
+##' 
+##' 
+
+
+download.NOAA_GEFS_downscale <- function(outfolder, lat.in, lon.in, sitename, start_date = Sys.time(), end_date = (as.POSIXct(start_date, tz="UTC") + lubridate::days(16)),
+                                         overwrite = FALSE, verbose = FALSE, ...) {
+  
+  start_date <- as.POSIXct(start_date, tz = "UTC")
+  end_date <- as.POSIXct(end_date, tz = "UTC")
+  
+  #It takes about 2 hours for NOAA GEFS weather data to be posted.  Therefore, if a request is made within that 2 hour window,
+  #we instead want to adjust the start time to the previous forecast, which is the most recent one avaliable.  (For example, if
+  #there's a request at 7:00 a.m., the data isn't up yet, so the function grabs the data at midnight instead.)
+  if (abs(as.numeric(Sys.time() - start_date, units="hours")) <= 2) {
+    start_date = start_date - lubridate::hours(2)
+    end_date = end_date - lubridate::hours(2)
+  }
+  
+  #Date/time error checking - Checks to see if the start date is before the end date
+  if (start_date > end_date) {
+    PEcAn.logger::logger.severe("Invalid dates: end date occurs before start date")
+  } else if (as.numeric(end_date - start_date, units="hours") < 6) { #Done separately to produce a more helpful error message.
+    PEcAn.logger::logger.severe("Times not far enough appart for a forecast to fall between them.  Forecasts occur every six hours; make sure start 
+                                and end dates are at least 6 hours appart.")
+  }
+  
+  #Set the end forecast date (default is the full 16 days)
+  if (end_date > start_date + lubridate::days(16)) {
+    end_date = start_date + lubridate::days(16)
```

Comment:
This should probably throw an informative warning that includes the updated `end_date`.